applications/graphs/urls.py

```diff
@@ -6,6 +6,7 @@
 
 	url(r'^graphs/$', views.graphs_page, name='graphs'),
 	url(r'^graphs/(?P<graph_id>[^/]+)$', views.graph_page, name='graph'),
+	url(r'^graph/(?P<graph_id>[^/]+)$', views.graph_page, name='graph_by_id'),
```

Comment:
Need that for a redirect when you click on the graph upload notification. It reduces the need for an IF statement as we can create an URL directly from notification table entry. (the word 'graph' is saved as a 'resource'). This also explains why we have a URL with 'group' and 'groups' (see comment below)